modules/sor/sorV2/lib/static.ts

```diff
@@ -32,6 +32,13 @@ export async function sorGetPathsWithPools(
     const basePools: BasePool[] = [];
 
     for (const prismaPool of prismaPools) {
+        // typeguard
+        if (prismaPool.protocolVersion === 3) {
+            if (!isLiquidityManagement(prismaPool.liquidityManagement)) {
```

Comment:
Moving the typeguard here seems to be more elegant as it just does not consider the pool for route building and simply continues.

Reply:
Yeah, the main problem was that it was throwing an error if LiquidityManagement wasn't properly set. 
This means that the whole SOR will break with a single corrupted pool in the DB.
Moving here allows us to simply skip that pool and later investigate why it doesn't have that attribute properly set